src/coreclr/nativeaot/Runtime/stressLog.cpp

```diff
@@ -132,25 +132,21 @@ void StressLog::Initialize(unsigned facilities,  unsigned level, unsigned maxByt
 /* create a new thread stress log buffer associated with pThread                 */
 
 ThreadStressLog* StressLog::CreateThreadStressLog(Thread * pThread) {
-
     if (theLog.facilitiesToLog == 0)
         return NULL;
 
     if (pThread == NULL)
         pThread = ThreadStore::GetCurrentThread();
 
+    //if we are not allowed to allocate stress log, we should not even try to take the lock
+    _ASSERTE(!pThread->IsInForbidBlockingRegion());
```

Comment:
This part a bit questionable. When we have a thread in asynchronously interrupted state, we should not do any blocking calls since we do not know what kind of locks the interrupted thread could be holding. 

In theory we need these asserts anyplace where we allocate, do IO or take locks. In practice the "unsafe" state is now very short - it is between the time we suspended the thread and the time we figure if it is in managed code (then we are safe) or not in managed (then we release the thread as not interesting to us). 

Basically we cannot block on the "IsManaged" path, which is very trivial now, so asserting all over the place seems an overkill and asserting only in stresslog seems strange.
Maybe just adding very clear comments on all methods involved would be sufficient?

Reply:
stresslog.cpp/stresslog.h was forked from coreclr for .NET Native UWP. Ideally, we would delete this copy of stresslog.cpp/stresslog.h and switched to using the one from CoreCLR utilcode again (add ifdef to the shared copy around things that are hard to unify).

I can see that some of the issues you have pointed out have been fixed in CoreCLR utilcode copy already.

Reply:
I mean more generally - if we track `forbidBlocking` state, we should check for it in a lot of places or not at all. How stresslog is different?

It looks like older code could be trying to do stackwalking without checking `IsManaged` and considered failed stackwalk as `IsManaged` check. That would involve a lot more code while in unsafe state, including stresslog, so the check in stresslog could be just a reaction to that pattern and the bugs that resulted from it.

Considering that the unsafe state is only relevant on the very isolated `IsManaged` code path (basically just an IP range check), I think tracking `forbidBlocking` state maybe not worth it entirely.

Reply:
I agree that this can be probably simplified.

Reply:
I've removed tracking of `forbidBlocking` and just added comments to the few affected methods that should not allocate or take runtime locks.